docs/modules/sql/pages/user-defined-types.adoc

```diff
@@ -1,60 +1,47 @@
 = User-Defined Types in SQL
 
-User-Defined Types (also known as UDTs, nested types, or nested fields) is an experimental feature that allows you to create
- custom data types that can be referenced in the xref:sql:create-mapping.adoc[CREATE MAPPING statement].
+User-Defined Types (also known as UDTs, nested types, or nested fields) allows you to create custom data types that can be referenced in the xref:sql:create-mapping.adoc[CREATE MAPPING statement].
 
-UDTs are useful for creating and accessing hierarchical data structures, including simple cases of nested objects and
-more complex cases of fully/partially connected graphs of objects. See <<queryingExamples,examples>> for more information.
-
-WARNING: Support for UDTs is an experimental feature, and disabled by default.
-See <<enabling-udt-support, Enabling UDT support>> to learn how to enable it.
+UDTs are useful for creating and accessing hierarchical data structures, which can even be cyclic. See <<queryingExamples,examples>> for more information.
 
 == Feature Overview
 
-Due to the experimental nature of UDTs, the feature set is limited to the following:
-
-- Support for three formats - `portable`, `compact` and `java` with varying level of sub-feature support
-- Instance and type-level cycles are only supported for Java types.
-- Type formats (referred to as type kind or kind hereafter) can not be mixed, both between mappings and types themselves.
-For example, you cannot use Java type in another Portable type or Portable mapping.
-- Limited support for instance and type-level cycles; only for Java types.
-Note that INSERT/UPDATE are disabled for mappings that use type hierarchies that contain cycles.
-- Limited support for `INSERT` and `UPDATE` queries for mappings using UDTs; if a `TYPE` hierarchy contains cycles,
-`INSERT` and `UPDATE` statements are disabled for the mapping.
-- Support for using UDT-based projections in both normal `SELECT` projection lists, `WHERE` filters and in `JOIN` conditions.
+- All serialization formats (namely `portable`, `compact`, `java` and `avro`) are supported.
+- Cyclic definitions are only supported for Java-serialized structures. (Portable and Compact formats do not support cyclic schemas at all. Avro support is currently limited to acyclic schemas.) Note that `INSERT`/`UPDATE` queries are disabled for mappings that use cyclic structures.
+- Type options _always_ override mapping options. Specifically, if the mapping has only `__key` or `this` fields, and both the mapping and type define a schema (Portable class definition, Compact schema, Java class or Avro schema), the one defined by the type is used. This also means that, in such situations, as long as the type defines a schema, the mapping does not need to.
+- If type schema is not specified, it will be resolved from the parent structure, which may be a mapping or another type.
```

Comment:
"the parent structure must provide the schema in a mapping or another type" should be "it will be resolved from the parent structure, which may be a mapping or another type". The parent structure **is** a mapping or type; it cannot specify schema in a mapping or type.

Reply:
Amended suggestion